tests/test_bulk_tx_multiep/src/main.xc

```diff
@@ -1,47 +1,46 @@
-#include <xs1.h>
-#include <print.h>
-#include <stdio.h>
-#include "xud.h"
-#include "platform.h"
-#include "shared.h"
-
-#define XUD_EP_COUNT_OUT   5
-#define XUD_EP_COUNT_IN    8
 
-/* Endpoint type tables */
-XUD_EpType epTypeTableOut[XUD_EP_COUNT_OUT] = {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
-XUD_EpType epTypeTableIn[XUD_EP_COUNT_IN] =   {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, 
-                                               XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
+#define EP_COUNT_OUT       (4)
+#define EP_COUNT_IN        (8)
 
-#define PKT_LENGTH_START 10
-#define PKT_LENGTH_END 19
+#define PKT_LENGTH_START   (10)
+#define PKT_LENGTH_END     (19)
 
+#include "shared.h"
 
-int main()
+XUD_EpType epTypeTableOut[EP_COUNT_OUT] = {XUD_EPTYPE_CTL,
+                                                XUD_EPTYPE_BUL,
+                                                 XUD_EPTYPE_BUL,
+                                                 XUD_EPTYPE_BUL,
+                                                 XUD_EPTYPE_BUL};
+XUD_EpType epTypeTableIn[EP_COUNT_IN] =   {XUD_EPTYPE_CTL, 
+                                                XUD_EPTYPE_BUL,
+                                                XUD_EPTYPE_BUL,
+                                                XUD_EPTYPE_BUL,
+                                                XUD_EPTYPE_BUL,
+                                                XUD_EPTYPE_BUL,
+                                                XUD_EPTYPE_BUL,
+                                                XUD_EPTYPE_BUL};
+
+unsigned test_func(chanend c_ep_out[EP_COUNT_OUT], chanend c_ep_in[EP_COUNT_IN])
 {
-    chan c_ep_out[XUD_EP_COUNT_OUT], c_ep_in[XUD_EP_COUNT_IN];
+    unsigned fail[4];
 
     par
     {
-        
-        XUD_Main(c_ep_out, XUD_EP_COUNT_OUT, c_ep_in, XUD_EP_COUNT_IN,
-                                null, epTypeTableOut, epTypeTableIn,
-                                XUD_SPEED_HS, XUD_PWR_BUS);
-
-        TestEp_Tx(c_ep_in[3], 3, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_LOOP);
-        TestEp_Tx(c_ep_in[4], 4, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_LOOP);
-        TestEp_Tx(c_ep_in[5], 5, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_LOOP);
-
-        {
-            TestEp_Tx(c_ep_in[6], 6, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_DIE);
-            XUD_ep ep0 = XUD_InitEp(c_ep_out[0]);
-            XUD_Kill(ep0);
-            exit(0);
-        }
-
+        fail[0] = TestEp_Tx(c_ep_in[3], 3, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_DIE);
+        fail[1] = TestEp_Tx(c_ep_in[4], 4, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_DIE);
+        fail[2] = TestEp_Tx(c_ep_in[5], 5, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_DIE);
+        fail[3] = TestEp_Tx(c_ep_in[6], 6, PKT_LENGTH_START, PKT_LENGTH_END, RUNMODE_DIE);
     }
 
-    return 0;
+    for(size_t i = 0; i < 4; i++)
+        fail[0] |= fail[i];
```

Comment:
How does the engineer investigating a failure of this test determine which run of `TestEp_Tx()` failed?

Reply:
Fixed

Reply:
See previous comment